antlr4 Trees module missing in JS runtime

I am trying import Trees module from JS runtime, but get "Cannot find module 'antlr4/tree/Trees' or its corresponding type declarations" error. See below screenshot.

![Screenshot 2024-10-16 at 8 52 55 PM](https://github.com/user-attachments/assets/91e8180b-a17a-4bf7-9e25-5f11f6621026)

`
import Trees from "antlr4/tree/Trees";
`

Based on the source code [Trees.js](https://github.com/antlr/antlr4/blob/dev/runtime/JavaScript/src/antlr4/tree/Trees.js), I should be able to import use above syntax.

Can someone look into why this module was not packaged into the runtime?

Target runtime: JavaScript / Typescript
Runtime version: 4.13.2

Have you tried import Trees from antlr4 ?Envoyé de mon iPhoneLe 17 oct. 2024 à 03:02, Lawrence Zeng ***@***.***> a écrit :﻿
I am trying import Trees module from JS runtime, but get "Cannot find module 'antlr4/tree/Trees' or its corresponding type declarations" error. See below screenshot.
Screenshot.2024-10-16.at.8.52.55.PM.png (view on web)
import Trees from "antlr4/tree/Trees";
Based on the source code Trees.js, I should be able to import use above syntax.
Can someone look into why this module was not packaged into the runtime?
Target runtime: JavaScript / Typescript

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

--- a/runtime/JavaScript/src/antlr4/tree/Trees.d.ts
+++ b/runtime/JavaScript/src/antlr4/tree/Trees.d.ts
@@ -1,0 +1,15 @@
+import { Parser } from "../Parser";
+import { ParseTree } from "./ParseTree";
+import { Tree } from "./Tree";
+
+export default Trees;
+declare namespace Trees {
+    function toStringTree(tree: Tree, ruleNames: string[], recog: Parser): string;
+    function getNodeText(t: Tree, ruleNames: string[], recog: Parser): string;
+    function getChildren(t: Tree): Tree[];
+    function getAncestors(t: Tree): Tree[];
+    function findAllTokenNodes(t: ParseTree, ttype: number): ParseTree[];
+    function findAllRuleNodes(t: ParseTree, ruleIndex: number): ParseTree[];
+    function findAllNodes(t: ParseTree, index: number, findTokens: boolean): ParseTree[];
+    function descendants(t: ParseTree): ParseTree[];
+}

--- a/runtime/JavaScript/src/antlr4/tree/index.d.ts
+++ b/runtime/JavaScript/src/antlr4/tree/index.d.ts
@@ -5,4 +5,4 @@ export * from './ParseTree';
 export * from './ParseTreeListener';
 export * from './ParseTreeVisitor';
 export * from './ParseTreeWalker';
-
+export * as Trees from './Trees';

